pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotQueryRuleSets.java

```diff
@@ -43,6 +43,8 @@ private PinotQueryRuleSets() {
           CoreRules.FILTER_AGGREGATE_TRANSPOSE,
           // push filter through set operation
           CoreRules.FILTER_SET_OP_TRANSPOSE,
+          // create a project on top of an aggregate when the aggregate is on top of a join
+          PinotRuleUtils.PINOT_AGGREGATE_EXTRACT_PROJECT_RULE,
```

Comment:
this approach doesn't work on count(*) or count(1) as 
1. pinot datatypes are non-nullable for now (which I think we should fix)
2. COUNT(*) and COUNT(1) doesn't have operand so it cannot be extracted into a project, we might need a special rule unfortunately.